### ISSUE TITLE
dd-trace breaks apps using `AsyncLocalStorage`

### ISSUE THREAD

**Issue by Drew-Kimberly** [DataDog/dd-trace-js#2056]
The `2.7.0` release of `dd-trace` breaks apps using `AsyncLocalStorage` (only tested ALS, decent chance this affects other `async_hooks` implementations as well). The async storage loses its context and so `AsyncLocalStorage::getStore()` method returns `undefined`.

**Expected behaviour**
`dd-trace` should not affect existing usage of `AsyncLocalStorage`

**Actual behaviour**
Calling `require('dd-trace').init()` causes existing calls to `als.getStore()` to return `undefined`.

**Steps to reproduce**
Here is a basic repo-repo: https://github.com/Drew-Kimberly/dd-trace-als-repo
```js
require('dd-trace').init()

const { AsyncLocalStorage } = require('async_hooks')
const express = require('express')

const als = new AsyncLocalStorage()
const app = express()

app.use((req, _, next) => {
    als.run({ path: req.path }, next)
})

app.get('*', (_, res) => {
    const store = als.getStore()
    if (!store) {
        throw new Error('ALS store is undefined')
    }

    return res.json(store)
})
```

Removing the `dd-trace` initialization or downgrading to version `2.6.0` resolves the issue.

**Environment**

* **Operation system:** OSX
* **Node.js version:** 16.15.0
* **Tracer version:** 2.7.0
* **Agent version:** N/A
* **Relevant library versions:** Express@4.18.1

**Comment by Drew-Kimberly** [comment]
@rochdev looks like this PR is likely responsible, as noted by this comment https://github.com/DataDog/dd-trace-js/pull/1995#issuecomment-1123850014

**Comment by rochdev** [comment]
Turns out we don't actually need the new AsyncResource instances that were added in the above PR for Web frameworks. A fix will be in the next release.

**Comment by Drew-Kimberly** [comment]
Wonderful news, thanks @rochdev!

**Comment by rochdev** [comment]
The fix for this was released in 2.8.0.

**Comment by mtheunissen82** [comment]
Hi, this issue still persists for Fastify 3.29.0.
step to reproduce are here https://github.com/mtheunissen82/als-dd-test

```js
require("dd-trace").init();

const { AsyncLocalStorage } = require("async_hooks");
const fastify = require("fastify");

const als = new AsyncLocalStorage();
const app = fastify();

app.addHook("onRequest", (request, reply, done) => {
  const requestContext = { path: request.url };
  als.run(requestContext, done);
});

app.get("/", (request, reply) => {
  const store = als.getStore();
  if (!store) {
    throw new Error("ALS store is undefined");
  }

  return reply.send(store);
});

app.listen(3000, (err, address) => {
  if (err) {
    console.error(err);
    process.exit(1);
  }
});
```
